core/controllers/editor.py

```diff
@@ -960,30 +962,43 @@ def post(self, exploration_id):
                 'Invalid filename extension: it should have '
                 'one of the following extensions: %s' % allowed_formats)
 
-        temp_suffix = '.' + extension
-        with tempfile.NamedTemporaryFile(suffix=temp_suffix) as f:
-            f.write(raw)
-            try:
-                audio = mutagen.File(f.name)
-            except mutagen.MutagenError:
-                raise self.InvalidInputException('Audio not recognized '
-                                                 'as a %s file' % extension)
-            f.close()
-
+        tempbuffer = StringIO.StringIO()
+        tempbuffer.write(raw)
```

Comment:
Here, do tempbuffer.write(copy.deepcopy(raw)), and then I think you won't need the "del audio".